dimod/reference/samplers/exact_solver.py

```diff
@@ -199,6 +200,76 @@ def sample_dqm(self, dqm, **kwargs):
         
         response = SampleSet.from_samples(possible_samples, 'DISCRETE', energies)
         return response
+    
+    
+class ExactCQMSolver():
+    """A simple exact solver for testing and debugging code using your local CPU.
+
+    Notes:
+        This solver calculates the energy for every possible
+        combination of variable cases. It becomes slow very quickly
+        
+    Examples:
+        This example solves a CQM with 3 variables and 1 constraint.
+        
+        >>> from dimod import ConstrainedQuadraticModel, Binary
+        >>> cqm = ConstrainedQuadraticModel()
+        >>> x, y, z = Binary('x'), Binary('y'), Binary('z')
+        >>> cqm.set_objective(x*y + 2*y*z)
+        >>> cqm.add_constraint(x*y == 1)
+    
+        >>> sampleset = ExactCQMSolver().sample_cqm(cqm)
+        >>> print(sampleset)
+          x y z energy num_oc. is_sat. is_fea.
+        0 0 0 0    0.0       1 arra...   False
+        1 0 1 0    0.0       1 arra...   False
+        2 1 0 0    0.0       1 arra...   False
+        4 0 0 1    0.0       1 arra...   False
+        6 1 0 1    0.0       1 arra...   False
+        3 1 1 0    1.0       1 arra...    True
+        5 0 1 1    2.0       1 arra...   False
+        7 1 1 1    3.0       1 arra...    True
+        ['INTEGER', 8 rows, 8 samples, 3 variables]
+        
+    """
+    properties = None
+    parameters = None
+
+    def __init__(self):
+        self.properties = {}
+        self.parameters = {}
+        
+    def sample_cqm(self, cqm, rtol: float = 1e-6, atol: float = 1e-8, **kwargs):
```

Comment:
should this have a `-> SampleSet` ?

Reply:
The other exact solvers don't have this present; would it be preferable for them to?

Reply:
Added `-> SampleSet` to each of the `sample_{}` methods